src/app/leaves/components/LeaveActionRender.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useMemo } from "react";
+import ActionsMenu from "@/common/ActionMenu/ActionsMenu";
+import {
+  Info as InfoIcon,
+  Trash2 as TrashIcon,
+  Edit as EditIcon,
+  CheckCircle as CheckCircleIcon,
+  AlertCircle as AlertCircleIcon,
+} from "lucide-react";
+import { MemberRole } from "@/lib/constants";
+
+const loggedUser = {
+  name: "John Smith",
+  role: "member",
+};
+
+const LeaveTableActionRender = ({
+  leaveRequestStatus,
+}: {
+  leaveRequestStatus: string | undefined;
```

Comment:
This type seems to be wrong. I remember we have enum defined for this, right? Also, why would that be undefined?

Reply:
This won't be the only thing you'd be needing, you'd need id and other stuff as well.

Reply:
I have addressed this comment by using an enum for the status type and also added a `leaveRequestId` prop, which will be used when we implement the handlers.